src/Symfony/Bundle/FrameworkBundle/Test/CsrfHelperTrait.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Test;
+
+use Symfony\Bundle\FrameworkBundle\KernelBrowser;
+use Symfony\Component\Security\Csrf\CsrfToken;
+
+/**
+ * Provides methods to deal with CSRF tokens in a stateless container.
+ */
+trait CsrfHelperTrait
+{
+    use SessionHelperTrait;
+
+    private function getCsrfToken(KernelBrowser $client, string $tokenId): CsrfToken
+    {
+        return $this->callInRequestContext($client, function () use ($tokenId) {
+            return static::getContainer()->get('security.csrf.token_manager')->getToken($tokenId);
```

Comment:
Be careful about that. In Symfony 5.3+, this requires that the authenticated user is the same (and that the session is the same, which won't work fine when using Panther)

Reply:
I'm not sure to understand. This PR is about CsrfToken and is not (If I understand correctly), related to users.
I get that UserToken and Users vary with the firewall but the session remains the same, and Csrf token are not namespace with the firewall.

Did I missed something?